src/decoder.rs

```diff
@@ -22,7 +20,7 @@ pub enum Value {
     Seq(Vec<Value>),
     Mapped(Box<Value>, Box<Value>),
     Branch(usize, Box<Value>),
-    Format(Box<Format>),
+    Format(Box<Format>, RefCell<Option<Decoder>>),
```

Comment:
We can use `std::cell:OnceCell` instead, since this is a cached value that is initialized once and never mutated after.

Reply:
```suggestion
Format(Box<Format>, OnceCell<Decoder>),
```